File: frontend/src/api/mindmap.ts
// 思维导图接口：整图 CRUD。data 为前端维护的整图 JSON（节点/连线/视图变换）。
import { apiGet, apiPost, apiPut, apiDelete } from './request'

/** 思维导图整图数据结构（前后端一致的 JSON 契约） */
export interface MindMapNode {
  id: string
  text: string
  x: number
  y: number
  parentId: string | null
  collapsed?: boolean
  color?: string
}

export interface MindMapEdge {
  id: string
  source: string
  target: string
}

export interface MindMapView {
  scale: number
  tx: number
  ty: number
}

export interface MindMapData {
  nodes: MindMapNode[]
  edges: MindMapEdge[]
  view: MindMapView
}

export interface MindMapSummary {
  id: number
  title: string
  createTime: string
  updateTime: string
}

export interface MindMapDetail {
  id: number
  userId: number
  title: string
  data: MindMapData
  createTime: string
  updateTime: string
}

export function listMindMaps() {
  return apiGet<MindMapSummary[]>('/mindmaps')
}

export function getMindMap(id: number) {
  return apiGet<MindMapDetail>(`/mindmaps/${id}`)
}

export function createMindMap(payload: { title: string; data?: MindMapData }) {
  return apiPost<number>('/mindmaps', payload)
}

export function updateMindMap(id: number, payload: { title?: string; data?: MindMapData }) {
  return apiPut<void>(`/mindmaps/${id}`, payload)
}

export function deleteMindMap(id: number) {
  return apiDelete<void>(`/mindmaps/${id}`)
}

File: backend/src/main/java/com/knowflow/entity/MindMap.java
package com.knowflow.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 学习工作台·思维导图（MindMap）：整图以 JSON 形式持久化在 data 列。
 * data 结构：{ nodes:[{id,text,x,y,parentId,collapsed,color}], edges:[{id,source,target}], view:{scale,tx,ty} }。
 * 层级关系通过节点的 parentId 表达，自由连线通过 edges 表达（见前端画布交互）。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("wb_mindmap")
public class MindMap extends BaseEntity {

    /** 所属用户ID（逻辑外键 sys_user.id） */
    private Long userId;

    /** 思维导图标题 */
    private String title;

    /** 整图数据（JSON 文本）：节点 / 连线 / 视图变换 */
    private String data;

    /** 归属知识库/分类ID（逻辑外键 doc_category.id，可空） */
    private Long categoryId;
}
